### PR TITLE
A Guide on How to Impute Time Series Missing Values using Python

### LINKED ISSUE #5904: A Complete Guide on How to Impute Time Series Missing Values in Python
#### Proposed title of article
A Complete Guide on How to Impute Time Series Missing Values in Python

#### Proposed article introduction
Time series is data that has a time component in it. In other words, the data collected sequentially over time is called time series. Usually, sometimes this data contain missing values, i.e., no record made at a specific time. The approach used to handle missing values in the Time series is usually different from those used to handle missing values in machine learning databases. In time series, special techniques that preserve the consistency of its components are used. 

Usually, we can deal with the missing values in time series data in two ways.

1. omit the entire instance that contains the missing value.
1. Impute the missing values using the method for time series imputations.

Imputation these values, we use methods such as:
1. Mean Imputation
1. Last Observation Carried Forward(LOCF)
1. Seasonality Interpolation
1. Linear interpolation

Intuitively, in this tutorial, we will discuss how each of the above methods works, and with facts, we'll declare which of them works the best. Along each method, we will provide its python implementation..

#### Key takeaways
 
The learn will clearly know:
1. What's time series.
1. Types of time series models.
1. What are time series missing values.
1. How to handle missing values in time series data.
1. Different methods for imputing time series missing values.
1. How to impute time series missing values in python

#### Article quality
This article will provide an in-depth approach to handling time series missing values. We will explore all techniques the learner must know to impute time series data. As all methods will not be applied simultaneously with the future time series, we will, together with the learner, figure out which method is the best to serve as a multipurpose technique with most of the times series. So, we will engage the learner to a high degree.

For a method being multipurpose, it doesn't mean the rest of the techniques are irrelevant. The learner needs to be provided with complete knowledge to have more flexibility on which method to use while handling their time-series data. They should be taught why we go for one method and not the other. So, this article will provide the learner with in-depth information on this topic.

As we will clearly illustrate all our discussions, our implementation session will ensure every code snippet is explained to the point, i.e., what exactly it's performing. So, to will make this article reader-friendly as much as we can.
#### References
N/A

### PR DESCRIPTION
**Pull request by DennisKimt**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5904

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by wilkistermumbi** [comment]
@ahmadmardeni1 I'd like to review this PR. 

cc: @hectorkambow

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

**Comment by ahmadmardeni1** [comment]
Hey @wilkistermumbi I'm sorry for the late reply. Can you choose another PR?

File: content/articles/missing-values-in-time-series/index.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+1. **Missing not at Random(MNAR):** The probability of missingness, in this case, depends on the variable itself.
+
+Time series models work with complete data, and therefore they require the missing data to be replaced with meaningful values before actual analysis. At a high level, missing values in time series are handled in two ways, either dropping them or replacing them. However, dropping missing values can be an inappropriate solution due to the time order of the data and the correlation between observations in adjacent periods.
+
+Estimating a reasonable value such that the components of the series are not distorted is a great approach to dealing with missing values in time series. The process of replacing missing values with values estimated from the same data or observed from the environment with the same conditions underlying the missing data is called *Imputation*.
+
+In python, let's look at various *imputation techniques* used in time series.
+
+### Python Implementation
```
Comment (BrianaNzivu) [inline content/articles/missing-values-in-time-series/index.md]: Let's have the steps clearly highlighted and well explained. By that I mean, 
Step one: ....

File: content/articles/missing-values-in-time-series/index.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
+### A Guide to Handle Time Series Missing Values
+When working with real-world data, it's a common challenge to encounter missing values. This article will guide us in addressing such a problem in time series data.
+
```
Comment (BrianaNzivu) [inline content/articles/missing-values-in-time-series/index.md]: Maybe you can add the table of contents to this article here.

File: content/articles/missing-values-in-time-series/index.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+1. **Missing not at Random(MNAR):** The probability of missingness, in this case, depends on the variable itself.
+
+Time series models work with complete data, and therefore they require the missing data to be replaced with meaningful values before actual analysis. At a high level, missing values in time series are handled in two ways, either dropping them or replacing them. However, dropping missing values can be an inappropriate solution due to the time order of the data and the correlation between observations in adjacent periods.
+
+Estimating a reasonable value such that the components of the series are not distorted is a great approach to dealing with missing values in time series. The process of replacing missing values with values estimated from the same data or observed from the environment with the same conditions underlying the missing data is called *Imputation*.
+
+In python, let's look at various *imputation techniques* used in time series.
+
+### Python Implementation
+In this session, we will work with the following libraries.
```
Comment (BrianaNzivu) [inline content/articles/missing-values-in-time-series/index.md]: Maybe you can list down the libraries.

File: content/articles/missing-values-in-time-series/index.md [hunk 4]
```diff
@@ -0,0 +1,258 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /missing-values-in-time-series/
+title: A Complete guide on how to impute Time Series Missing Values in Python
+description: This tutorial will cover how to impute missing values in a time series in Python.
+author:
```
Comment (BrianaNzivu) [inline content/articles/missing-values-in-time-series/index.md:8]: Please link your author folder's name here.

**Review by BrianaNzivu — CHANGES_REQUESTED** [review]
Hey @DennisKimt  👋 .

Good job on the article. It is a very interesting read. Barely any grammar changes to be made.
This is very impressive. Keep up the good work 👍 .

Just a few changes I need you to make as we continue with the preliminary revisions:
- Please structure your article according to this [sample](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md). I need you to clearly highlight and explain your steps. Where you can add more detail please do so.
- Please refer to the comment that I have made.

I would also love to know if this is your first article?

In case of any question or help needed please do not hesitate to ping me here or on Slack.
Also, notify me when you are done.

Regards.

**Comment by BrianaNzivu** [comment]
Proceeding with tech review meanwhile.

**Comment by BrianaNzivu** [comment]
Tech review done 👍

**Comment by BrianaNzivu** [comment]
@DennisKimt 👋 .
How's the progress with the corrections?

**Review by BrianaNzivu — CHANGES_REQUESTED** [review]
Hey @DennisKimt 👋 .
You haven't made any of the changes I requested except linking your author name. Please refer to all the comments I made.

Ping me when done.
Regards.

**Comment by DennisKimt** [comment]
@BrianaNzivu  I've completed the changes you requested

**Review by BrianaNzivu — APPROVED** [review]
Hey @DennisKimt  👋 .

Very good job on the article. It is a very interesting read 👏 .

Just a few things I'd like to point out:
- I suggest that you check out the [structure](https://github.com/section-engineering-education/engineering-education/tree/master/new_contributors) for the various articles accepted by Section.
-  It is always advisable to submit an article that is well polished. By this, I mean having an article that does not have too many grammatical corrections and functional code. I suggest that you check out our [resources](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) page which has links to various resourceful tools for writing. 
- Lastly please go through ALL the corrections I have made so that you can not repeat the same mistakes.

That's it! :smile: 

A final review will be done now by a content moderator.
Otherwise, congratulations on publishing your article🍾.

In case of any question or clarification (currently or in the future), do not hesitate to reach out via [Slack](https://app.slack.com/client/T5X14SH7G/D01A6MT673R) or my personal email (briananzivu@gmail.com). 
I am **ALWAYS** here to help.

It has been a pleasure reading and reviewing this article.

Looking forward to seeing you contribute more to EngEd. 🚀

Kind Regards.

**Comment by lalith1403** [comment]
Jumping on this one 🚀

**Comment by hectorkambow** [comment]
@DennisKimt Please reach out via DM on slack

**Pull request merged**